### DISCUSSION TITLE
Do paths.types apply to all paths between all TXs and RXs?

### DISCUSSION THREAD

**Discussion by jmoraispk** [NVlabs/sionna#203] [Q&A] [answered]
Respected maintainers, 

In Sionna RT, paths.types is of dimensions [batch, max_num_paths]. Yet, say, the paths' delay is [batch_size, num_rx, num_tx, max_num_paths]. This suggests that the paths between all transmitters and receivers have the same types. But one can find two pairs of TX and RX positions with different number of reflected, diffracted or scattered paths. So either paths.types only refers to one pair of TX and RX, or the max_num_paths is set based on the maximum number of reflected, diffracted, and scattered paths across all positions such that the paths.types is common and applies for all pairs of tx-rx.

Could you confirm which one is it? Is there a third option?

I tried finding the answer in [paths' source](https://nvlabs.github.io/sionna/_modules/sionna/rt/paths.html), but couldn't decipher the part that starts with "# Merge types". Some help will be greatly appreciated!

**Comment by faycalaa** [comment, marked as answer]
The correct choice is the second option you pointed out.
